packages/backend/src/services/RenameService/rename.ts

```diff
@@ -684,6 +697,8 @@ export const renameSavedChart = ({
     if (validate)
         validateRename(chart, updatedChart, chart.name, 'chart', nameChanges);
 
+    // Only mark changes if there was actually an update. There are renameFunctions that won't change the values if not needed
+    const hasChanges = JSON.stringify(chart) !== JSON.stringify(updatedChart);
```

Comment:
Another false positive here. The `hasChanges` flag was always set to true whenever the replace function was called, regardless of whether it actually changed anything. We only want to change this flag if there was actually a change, so we do a final string-to-string comparison to set it instead.